### PR TITLE
Ignored Nan and Infinite float values

### PR DESCRIPTION
**Pull request by jorik**
These values cannot be handled by NRDB, and POMI currently logs an error message for every single NaN/Infinite value it processes, which can be confusing.

### CODE REVIEW

File: internal/integration/harvester_decorator.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	}
+}
+
+func (ha harvesterDecorator) HarvestNow(ctx context.Context) {
+	ha.innerHarvester.HarvestNow(ctx)
+}
+
+func (ha harvesterDecorator) processMetric(f float64, m telemetry.Metric) {
+	if isNaNOrInfinity(f) {
+		logrus.Debugf("Ignoring NaN float value for metric: %v", m)
```
Comment (paologallinaharbur) [inline internal/integration/harvester_decorator.go:36]: it could be also ±inf

File: internal/integration/emitter.go [hunk 2]
```diff
@@ -161,14 +168,19 @@ func NewTelemetryEmitter(cfg TelemetryEmitterConfig) (*TelemetryEmitter, error)
...
-	harvester, err := telemetry.NewHarvester(cfg.HarvesterOpts...)
+	var h harvester
+	h, err := telemetry.NewHarvester(cfg.HarvesterOpts...)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not create new Harvester")
 	}
 
+	// Wrap the harvester so we can filter out invalid float values: NaN and Infinity.
+	// If we do send them, the harvester will always output these as errors
+	h = harvesterDecorator{h}
```
Comment (paologallinaharbur) [inline internal/integration/emitter.go:179]: nice approach!

**Review by paologallinaharbur — APPROVED** [review]
LGTM! thanks for the change!

File: internal/integration/harvester_decorator.go (hunk 1 above)
Comment (jorik) [reply internal/integration/harvester_decorator.go:36]: fixed!

**Review by paologallinaharbur — APPROVED** [review]
LGTM, Thanks for the update!

**Pull request merged**